chain/pool/txpool.go

```diff
@@ -34,27 +34,27 @@ var (
 
 // TxnPool is a list of txns that need to by add to ledger sent by user.
 type TxnPool struct {
-	mu          sync.RWMutex
-	TxLists     map[common.Uint160]*NonceSortedTxs // NonceSortedTxs instance to store user's account.
-	SigChainTxs map[common.Uint256]*Transaction    // tx with sigchain type.
-
+	TxLists sync.Map // NonceSortedTxs instance to store user's account.
 }
 
 func NewTxPool() *TxnPool {
-	return &TxnPool{
-		TxLists:     make(map[common.Uint160]*NonceSortedTxs),
-		SigChainTxs: make(map[common.Uint256]*Transaction),
-	}
+	return &TxnPool{}
 }
 
 func (tp *TxnPool) AppendTxnPool(txn *Transaction) error {
-	//1. process all Orphens
-	for _, list := range tp.TxLists {
-		list.ProcessOrphans(tp.processTx)
-	}
-	for _, list := range tp.TxLists {
-		list.CleanOrphans(nil)
-	}
+	// 1. process all Orphens
+	tp.TxLists.Range(func(_, v interface{}) bool {
+		if list, ok := v.(*NonceSortedTxs); ok {
+			list.ProcessOrphans(tp.processTx)
+		}
+		return true
+	})
+	tp.TxLists.Range(func(_, v interface{}) bool {
```

Comment:
Why traversed tp.TxLists twice?  Would tp.TxLists iterator be influenced by v.CleanOrphans?
What happen if
```
tp.TxLists.Range( 
    func(_, v interface{}) bool {
        ......
        v.ProcessOrphans(...)
        v.CleanOrphans(nil)
        ......
        return true
    }
)
```